<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ scFlow</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scFlow</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.7.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/scFlow.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/combiz/scFlow/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="import-and-export">Import and Export <a href="#import-and-export" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="read_celltype_mappings.html">read_celltype_mappings()</a></code> </p>
        </td>
        <td><p>Read Celltype Mappings</p></td>
      </tr><tr><td>
          <p><code><a href="read_sce.html">read_sce()</a></code> </p>
        </td>
        <td><p>Read a SingleCellExperiment from a folder</p></td>
      </tr><tr><td>
          <p><code><a href="read_sparse_matrix.html">read_sparse_matrix()</a></code> </p>
        </td>
        <td><p>Read a feature-barcode matrix</p></td>
      </tr><tr><td>
          <p><code><a href="write_celltype_mappings.html">write_celltype_mappings()</a></code> </p>
        </td>
        <td><p>Write Celltype Mappings</p></td>
      </tr><tr><td>
          <p><code><a href="write_sce.html">write_sce()</a></code> </p>
        </td>
        <td><p>Write a SingleCellExperiment to a folder</p></td>
      </tr><tr><td>
          <p><code><a href="write_sparse_matrix.html">write_sparse_matrix()</a></code> </p>
        </td>
        <td><p>Write a feature-barcode matrix</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="pre-processing-and-annotation">Pre-Processing and Annotation <a href="#pre-processing-and-annotation" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="annotate_celltype_metrics.html">annotate_celltype_metrics()</a></code> </p>
        </td>
        <td><p>Annotate a SingleCellExperiment With Cell-type Metrics</p></td>
      </tr><tr><td>
          <p><code><a href="annotate_integrated_sce.html">annotate_integrated_sce()</a></code> </p>
        </td>
        <td><p>Annotate integrated, reduced dimension,
and clustered SingleCellExperiment object</p></td>
      </tr><tr><td>
          <p><code><a href="annotate_merged_sce.html">annotate_merged_sce()</a></code> </p>
        </td>
        <td><p>Annotate a post-merge SingleCellExperiment with plots</p></td>
      </tr><tr><td>
          <p><code><a href="annotate_sce.html">annotate_sce()</a></code> </p>
        </td>
        <td><p>Annotate a SingleCellExperiment With Gene Names and QC Metrics</p></td>
      </tr><tr><td>
          <p><code><a href="annotate_sce_cells.html">annotate_sce_cells()</a></code> </p>
        </td>
        <td><p>Add basic cell-wise annotations for a SingleCellExperiment</p></td>
      </tr><tr><td>
          <p><code><a href="annotate_sce_genes.html">annotate_sce_genes()</a></code> </p>
        </td>
        <td><p>Add basic gene-wise annotations for a SingleCellExperiment</p></td>
      </tr><tr><td>
          <p><code><a href="filter_sce.html">filter_sce()</a></code> </p>
        </td>
        <td><p>Filter SingleCellExperiment according to QC metrics
Quality control metrics previously calculated are used to subset the
SingleCellExperiment for both cells and genes passing QC criteria previously
specified in the <code>annotate_sce_genes</code> and <code>annotate_sce_cells</code> functions</p></td>
      </tr><tr><td>
          <p><code><a href="find_cells.html">find_cells()</a></code> </p>
        </td>
        <td><p>Annotate a SingleCellExperiment with EmptyDrops predictions</p></td>
      </tr><tr><td>
          <p><code><a href="find_singlets.html">find_singlets()</a></code> </p>
        </td>
        <td><p>Find singlets in a SingleCellExperiment</p></td>
      </tr><tr><td>
          <p><code><a href="generate_sce.html">generate_sce()</a></code> </p>
        </td>
        <td><p>Generate a SingleCellExperiment object from a matrix and metadata</p></td>
      </tr><tr><td>
          <p><code><a href="map_ensembl_gene_id.html">map_ensembl_gene_id()</a></code> </p>
        </td>
        <td><p>Return mappings for Ensembl Gene IDs</p></td>
      </tr><tr><td>
          <p><code><a href="merge_sce.html">merge_sce()</a></code> </p>
        </td>
        <td><p>Merge Multiple SingleCellExperiment Objects</p></td>
      </tr><tr><td>
          <p><code><a href="read_metadata.html">read_metadata()</a></code> </p>
        </td>
        <td><p>Read the metadata for a sample from a samplesheet</p></td>
      </tr><tr><td>
          <p><code><a href="report_celltype_metrics.html">report_celltype_metrics()</a></code> </p>
        </td>
        <td><p>Generate A Report of Cell-Type Metrics for a SingleCellExperiment</p></td>
      </tr><tr><td>
          <p><code><a href="report_celltype_model.html">report_celltype_model()</a></code> </p>
        </td>
        <td><p>Generate a report for a celltype proportion analysis</p></td>
      </tr><tr><td>
          <p><code><a href="report_merged_sce.html">report_merged_sce()</a></code> </p>
        </td>
        <td><p>Generate plots and a QC report for a SingleCellExperiment</p></td>
      </tr><tr><td>
          <p><code><a href="report_qc_sce.html">report_qc_sce()</a></code> </p>
        </td>
        <td><p>Generate plots and a QC report for a SingleCellExperiment</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="clustering-and-dimensionality-reduction">Clustering and Dimensionality Reduction <a href="#clustering-and-dimensionality-reduction" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="cluster_sce.html">cluster_sce()</a></code> </p>
        </td>
        <td><p>Cluster SingleCellExperiment with monocle3::cluster_cells</p></td>
      </tr><tr><td>
          <p><code><a href="map_custom_celltypes.html">map_custom_celltypes()</a></code> </p>
        </td>
        <td><p>Map Custom Celltype Annotations from a DataFrame</p></td>
      </tr><tr><td>
          <p><code><a href="reduce_dims_sce.html">reduce_dims_sce()</a></code> </p>
        </td>
        <td><p>Calculate dimensionality reductions for a SingleCellExperiment object
or merged SingleCellExperiment objects using tSNE, UMAP, UMAP3D</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="differential-gene-expression">Differential Gene Expression <a href="#differential-gene-expression" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="perform_de.html">perform_de()</a></code> </p>
        </td>
        <td><p>Perform Differential Gene Expression on a SingleCellExperiment</p></td>
      </tr><tr><td>
          <p><code><a href="pseudobulk_sce.html">pseudobulk_sce()</a></code> </p>
        </td>
        <td><p>Generate a Pseudo-bulk SingleCellExperiment from a SingleCellExperiment</p></td>
      </tr><tr><td>
          <p><code><a href="report_de.html">report_de()</a></code> </p>
        </td>
        <td><p>Generate a report for differential expression analysis</p></td>
      </tr><tr><td>
          <p><code><a href="volcano_plot.html">volcano_plot()</a></code> </p>
        </td>
        <td><p>Plot volcano plot for differential expression analysis</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="data-visualization">Data Visualization <a href="#data-visualization" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="plot_expr_by_numeric_var.html">plot_expr_by_numeric_var()</a></code> </p>
        </td>
        <td><p>Plot Gene Expression for Samples against Numerical Variable</p></td>
      </tr><tr><td>
          <p><code><a href="plot_reduced_dim.html">plot_reduced_dim()</a></code> </p>
        </td>
        <td><p>Plot Cells Reduced Dimensions With Feature Highlighting</p></td>
      </tr><tr><td>
          <p><code><a href="plot_reduced_dim_gene.html">plot_reduced_dim_gene()</a></code> </p>
        </td>
        <td><p>Plot Cells Reduced Dimensions With Gene Expression</p></td>
      </tr><tr><td>
          <p><code><a href="plot_violin.html">plot_violin()</a></code> </p>
        </td>
        <td><p>Plot Gene Expression Violin Plots Stratified by Group</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="impacted-pathway-analysis">Impacted Pathway Analysis <a href="#impacted-pathway-analysis" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="find_impacted_pathways.html">find_impacted_pathways()</a></code> </p>
        </td>
        <td><p>Functional enrichment analysis</p></td>
      </tr><tr><td>
          <p><code><a href="list_databases.html">list_databases()</a></code> </p>
        </td>
        <td><p>Check available databases and their alias for scFlow</p></td>
      </tr><tr><td>
          <p><code><a href="pathway_analysis_enrichr.html">pathway_analysis_enrichr()</a></code> </p>
        </td>
        <td><p>Functional enrichment analysis using enrichR</p></td>
      </tr><tr><td>
          <p><code><a href="pathway_analysis_webgestaltr.html">pathway_analysis_webgestaltr()</a></code> </p>
        </td>
        <td><p>Functional enrichment analysis using WebgestaltR</p></td>
      </tr><tr><td>
          <p><code><a href="report_impacted_pathway.html">report_impacted_pathway()</a></code> </p>
        </td>
        <td><p>Generate a report for impacted pathway analysis</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="data-integration">Data Integration <a href="#data-integration" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="integrate_sce.html">integrate_sce()</a></code> </p>
        </td>
        <td><p>Integrate datasets within a SingleCellExperiment Object</p></td>
      </tr><tr><td>
          <p><code><a href="liger_preprocess.html">liger_preprocess()</a></code> </p>
        </td>
        <td><p>Preprocessing steps for Liger dimensionality reduction</p></td>
      </tr><tr><td>
          <p><code><a href="liger_reduce_dims.html">liger_reduce_dims()</a></code> </p>
        </td>
        <td><p>Dimensionality reduction using Liger factorization</p></td>
      </tr><tr><td>
          <p><code><a href="report_integrated_sce.html">report_integrated_sce()</a></code> </p>
        </td>
        <td><p>Generate a report for dataset integration, dims reduction, and clustering</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="celltype-annotation">Celltype annotation <a href="#celltype-annotation" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="find_marker_genes.html">find_marker_genes()</a></code> </p>
        </td>
        <td><p>Find specific markers for groups of cells</p></td>
      </tr><tr><td>
          <p><code><a href="map_celltypes_sce.html">map_celltypes_sce()</a></code> </p>
        </td>
        <td><p>Map cluster celltypes with EWCE</p></td>
      </tr><tr><td>
          <p><code><a href="model_celltype_freqs.html">model_celltype_freqs()</a></code> </p>
        </td>
        <td><p>Model Changes in Celltype Frequencies with Dirichlet Multinomial Regression</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Combiz Khozoie, Nurun Fancy, Mahdi Marjaneh, Michael Thomas, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

